### DISCUSSION TITLE
Restore BM25 model

### DISCUSSION THREAD

**Discussion by landonbar** [winkjs/wink-nlp#63] [Help on coding] [answered]
Is it possible to restore a BM25 model from the its.modelJSON output so it could vectorize without relearning?

**Comment by sanjayaksaxena** [comment, marked as answer]
Currently this capability is not available. If there are strong use-cases then we can consider addition of such a feature.

**Reply by landonbar** [reply]
In my case, I'm using docTermMatrix from the BM25 vectorizer as features to a SVM model. I would like to generate BM25 vectors from untrained documents based on the original training set and let the SVM classify the document. Because of the large corpus, I would not want to retrain the vectorizer each time I want to generate a vector for a new document and the shape needs to stay constant for the classifier.

**Reply by sanjayaksaxena** [reply]
Any thoughts on how would you like to see it working?

**Reply by landonbar** [reply]
Either an option to pass the modelJson output into the config at instantiation or a .load(modelJson) method on the vectorizer.

**Reply by sanjayaksaxena** [reply]
Sounds great, we decided to provide a separate api — `.loadModel( json )` to enable loading of saved models. It accepts `json` (i.e. string) as input and loads it. Once successfully loaded further learning is not possible. Released v1.10.0 with this feature, please check it out!

Will take a bit to update the documentation.

**Comment by landonbar** [comment]
I will. Thanks!

**Reply by sanjayaksaxena** [reply]
We have also updated the [BM25Vectorizer documentation](https://winkjs.org/wink-nlp/bm25-vectorizer.html).
